sys/arm64/include/cheri.h

```diff
@@ -40,7 +40,9 @@
 #endif
 
 #ifdef _KERNEL
-#define	__USER_DDC	((void * __capability)curthread->td_frame->tf_ddc)
+#define	__USER_DDC ((cheri_getperm(__USER_PCC) & CHERI_PERM_EXECUTIVE) ? \
```

Comment:
perhaps swap order of  __USER_PCC and __USER_DDC defintions

Reply:
This keeps them in the same order across architectures